enumerables.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Style/RedundantSelf, Style/GuardClause, Style/IfUnlessModifier, Metrics/BlockNesting, Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    i = 0
+    while i < array.length
+      if block_given?
+        yield array[i]
+      else
+        return "\#<Enumerator: #{self}:my_each>"
+      end
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    array = self if self.class == Array
+    array = self.to_a if self.class == Range
+    value = 0
+    while value < array.length
+      if block_given?
+        yield(array[value], value)
+      else
+        return "\#<Enumerator: #{self}:my_each_with_index>"
+      end
+      value += 1
+    end
+    array
+  end
+
+  def my_select
+    array = []
+    i = 0
+    while i < self.length
+      if yield self[i]
+        array.push(self[i])
+      end
+      i += 1
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield(self[i]) == false
+            return false
+          end
+        elsif self[i].nil?
+          return false
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      return false
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            return true
+          end
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      j = 0
+      while j < self.length
+        if args[0] == self[j]
+          return true
+        end
+
+        j += 1
+      end
+
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            return false
+          end
+        elsif self[i]
+          return false
+        end
+        i += 1
+      end
+    else
+      j = 0
+      while j < self.length
+        if self[j] == args[0]
+          return false
+        else
+          true
+        end
+
+        j += 1
+      end
+    end
+    true
+  end
+
+  def my_count(*num)
+    count = 0
+    if num.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            count += 1
+          end
+        else
+          return self.length
+        end
+        i += 1
+      end
+    elsif num.length == 1
+      i = 0
+      while i < self.length
+        if num[0] == self[i]
+          count += 1
+        end
+        i += 1
+      end
+    end
+    count
+  end
```

Comment:
- [ ] my_count returns the number of items in enum through enumeration in a range
(range.my_count) should be equal range.count